seed/seedwriter/writer.go

```diff
@@ -1225,9 +1223,67 @@ func (w *Writer) Downloaded(fetchAsserts AssertsFetchFunc) (complete bool, err e
 	if err := w.ensureARefs(nil, fetchAsserts); err != nil {
 		return false, err
 	}
+
+	if err := w.checkPrereqs(); err != nil {
+		return false, err
+	}
+
 	return true, nil
 }
 
+func (w *Writer) checkPrereqs() error {
+	modes := make([]string, 0, len(w.availableByMode))
+	modes = append(modes, "run")
+	fixed := 1
+	if _, ok := w.availableByMode["ephemeral"]; ok {
+		modes = append(modes, "ephemeral")
+		fixed = 2
+	}
+	for m := range w.availableByMode {
+		if m == "run" || m == "ephemeral" {
+			continue
+		}
+		modes = append(modes, m)
+	}
+	sort.Strings(modes[fixed:])
+	w.checkPrereqsInMode("ephemeral")
+	for _, m := range modes {
+		if err := w.checkPrereqsInMode(m); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (w *Writer) checkPrereqsInMode(mode string) error {
+	nmode := len(w.byModeSnaps[mode])
+	nephemeral := len(w.byModeSnaps["ephemeral"])
+	var snaps []*snap.Info
+	if mode != "run" && mode != "ephemeral" {
```

Comment:
So this would be either recovery or install for which ephemeral is an alias? Maybe worth adding a comment about this.

Reply:
or factory-reset. The code in writer tries not to assume a fixed list of modes